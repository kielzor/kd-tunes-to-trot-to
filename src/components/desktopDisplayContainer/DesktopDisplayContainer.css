.desktop-display {
  display: flex;
  width: 100%;
  align-items: center;
  height: 60vh;
  justify-content: center; }

.desktop-button-parent {
  display: flex;
  width: 70%;
  justify-content: space-between; }

.desktop-container-button {
  height: 12rem;
  width: 12rem; }
