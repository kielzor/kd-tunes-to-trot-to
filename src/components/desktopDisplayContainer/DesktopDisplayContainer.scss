.desktop-display {
  display: flex;
  width: 100%;
  flex-direction: column;
  align-items: center;
  height: 60vh;
  justify-content: center;
}

.desktop-button-parent {
  display: flex;
  width: 50%;
  align-items: center;
  justify-content: space-between;
  margin: 5rem auto;
}

.desktop-container-button {
  background-color: #3C7A89;
  height: 12rem;
  width: 12rem;
  border-radius: 10%;
  color: white;
}

.music-player-route {
  width: 8rem;
  height: 8rem;
  border-radius: 10%;
  background-color: #9FA2B2;
}

button, select {
  cursor: pointer;
}