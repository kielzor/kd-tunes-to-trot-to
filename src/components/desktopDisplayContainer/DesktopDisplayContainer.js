import React, { Component } from 'react'
import './DesktopDisplayContainer.css'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

class DesktopDisplayContainer extends Component {
	render() {
		return (
			<div className='desktop-display'>
				<div className='desktop-button-parent'>
					<Link 
						className='desktop-link-workout'
						to='/workout'
						onClick={this.props.handleDesktopClick}
					>
						<button className='desktop-container-button'>CREATE NEW WORKOUT</button>
					</Link>
					<Link 
						className='desktop-link-player'
						to='/music-player'
						onClick={this.props.handleDesktopClick}
					>
						<button className='music-player-route'>MusicPlayer</button>
					</Link>
					<Link 
						className='desktop-link-import'
						to='/import'
						onClick={this.props.handleDesktopClick}
					>
						<button className='desktop-container-button'>IMPORT TRACKS</button>
					</Link>
				</div>
			</div>
		)
	}
}

DesktopDisplayContainer.propTypes = {
	handleDesktopClick: PropTypes.func
}

export default DesktopDisplayContainer