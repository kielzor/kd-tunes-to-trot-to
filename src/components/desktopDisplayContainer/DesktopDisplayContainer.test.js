import DesktopDisplayContainer from './DesktopDisplayContainer'
import React from 'react'
import { shallow } from 'enzyme'

describe('DesktopDisplayContainer', () => {
  let wrapper
  let mockHandleDesktopClick = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<DesktopDisplayContainer handleDesktopClick={mockHandleDesktopClick} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call mock function on workout click', () => {
    wrapper.find('.desktop-link-workout').simulate('click')

    expect(mockHandleDesktopClick).toHaveBeenCalled()
  })

  it('should call mock function on player click', () => {
    wrapper.find('.desktop-link-player').simulate('click')

    expect(mockHandleDesktopClick).toHaveBeenCalled()
  })

  it('should call mock function on import click', () => {
    wrapper.find('.desktop-link-import').simulate('click')

    expect(mockHandleDesktopClick).toHaveBeenCalled()
  })
})