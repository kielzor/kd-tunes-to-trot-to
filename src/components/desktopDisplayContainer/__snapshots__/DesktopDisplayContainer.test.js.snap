// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopDisplayContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DesktopDisplayContainer
    handleDesktopClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="desktop-button-parent"
>
        <Link
                className="desktop-link-workout"
                onClick={[Function]}
                replace={false}
                to="/workout"
        >
                <button
                        className="desktop-container-button"
                >
                        CREATE NEW WORKOUT
                </button>
        </Link>
        <Link
                className="desktop-link-player"
                onClick={[Function]}
                replace={false}
                to="/music-player"
        >
                <button
                        className="music-player-route"
                >
                        MusicPlayer
                </button>
        </Link>
        <Link
                className="desktop-link-import"
                onClick={[Function]}
                replace={false}
                to="/import"
        >
                <button
                        className="desktop-container-button"
                >
                        IMPORT TRACKS
                </button>
        </Link>
</div>,
      "className": "desktop-display",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="desktop-link-workout"
            onClick={[Function]}
            replace={false}
            to="/workout"
>
            <button
                        className="desktop-container-button"
            >
                        CREATE NEW WORKOUT
            </button>
</Link>,
          <Link
            className="desktop-link-player"
            onClick={[Function]}
            replace={false}
            to="/music-player"
>
            <button
                        className="music-player-route"
            >
                        MusicPlayer
            </button>
</Link>,
          <Link
            className="desktop-link-import"
            onClick={[Function]}
            replace={false}
            to="/import"
>
            <button
                        className="desktop-container-button"
            >
                        IMPORT TRACKS
            </button>
</Link>,
        ],
        "className": "desktop-button-parent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="desktop-container-button"
>
              CREATE NEW WORKOUT
</button>,
            "className": "desktop-link-workout",
            "onClick": [Function],
            "replace": false,
            "to": "/workout",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CREATE NEW WORKOUT",
              "className": "desktop-container-button",
            },
            "ref": null,
            "rendered": "CREATE NEW WORKOUT",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="music-player-route"
>
              MusicPlayer
</button>,
            "className": "desktop-link-player",
            "onClick": [Function],
            "replace": false,
            "to": "/music-player",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MusicPlayer",
              "className": "music-player-route",
            },
            "ref": null,
            "rendered": "MusicPlayer",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="desktop-container-button"
>
              IMPORT TRACKS
</button>,
            "className": "desktop-link-import",
            "onClick": [Function],
            "replace": false,
            "to": "/import",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "IMPORT TRACKS",
              "className": "desktop-container-button",
            },
            "ref": null,
            "rendered": "IMPORT TRACKS",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="desktop-button-parent"
>
          <Link
                    className="desktop-link-workout"
                    onClick={[Function]}
                    replace={false}
                    to="/workout"
          >
                    <button
                              className="desktop-container-button"
                    >
                              CREATE NEW WORKOUT
                    </button>
          </Link>
          <Link
                    className="desktop-link-player"
                    onClick={[Function]}
                    replace={false}
                    to="/music-player"
          >
                    <button
                              className="music-player-route"
                    >
                              MusicPlayer
                    </button>
          </Link>
          <Link
                    className="desktop-link-import"
                    onClick={[Function]}
                    replace={false}
                    to="/import"
          >
                    <button
                              className="desktop-container-button"
                    >
                              IMPORT TRACKS
                    </button>
          </Link>
</div>,
        "className": "desktop-display",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="desktop-link-workout"
              onClick={[Function]}
              replace={false}
              to="/workout"
>
              <button
                            className="desktop-container-button"
              >
                            CREATE NEW WORKOUT
              </button>
</Link>,
            <Link
              className="desktop-link-player"
              onClick={[Function]}
              replace={false}
              to="/music-player"
>
              <button
                            className="music-player-route"
              >
                            MusicPlayer
              </button>
</Link>,
            <Link
              className="desktop-link-import"
              onClick={[Function]}
              replace={false}
              to="/import"
>
              <button
                            className="desktop-container-button"
              >
                            IMPORT TRACKS
              </button>
</Link>,
          ],
          "className": "desktop-button-parent",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="desktop-container-button"
>
                CREATE NEW WORKOUT
</button>,
              "className": "desktop-link-workout",
              "onClick": [Function],
              "replace": false,
              "to": "/workout",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CREATE NEW WORKOUT",
                "className": "desktop-container-button",
              },
              "ref": null,
              "rendered": "CREATE NEW WORKOUT",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="music-player-route"
>
                MusicPlayer
</button>,
              "className": "desktop-link-player",
              "onClick": [Function],
              "replace": false,
              "to": "/music-player",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MusicPlayer",
                "className": "music-player-route",
              },
              "ref": null,
              "rendered": "MusicPlayer",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="desktop-container-button"
>
                IMPORT TRACKS
</button>,
              "className": "desktop-link-import",
              "onClick": [Function],
              "replace": false,
              "to": "/import",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "IMPORT TRACKS",
                "className": "desktop-container-button",
              },
              "ref": null,
              "rendered": "IMPORT TRACKS",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
